ext/src/http/client/curl/http_client_curl.cc

```diff
@@ -428,6 +441,11 @@ bool HttpClient::MaybeSpawnBackgroundThread()
 
   background_thread_.reset(new std::thread(
       [](HttpClient *self) {
+        if (self->background_thread_instrumentation_ != nullptr)
+        {
+          self->background_thread_instrumentation_->OnStart();
+        }
+
         int still_running = 1;
         std::chrono::system_clock::time_point last_free_job_timepoint =
             std::chrono::system_clock::now();
```

Comment:
Do we need to call `BeforeWait` and `AfterWait` before and after `curl_multi_poll` or `curl_multi_wait`? These calls may block until an event occurs or a timeout is reached.

Reply:
Yes, good point. Implemented.